cmake/Abseil.cmake

```diff
@@ -0,0 +1,56 @@
+macro(p4c_obtain_abseil)
+  option(
+    P4C_USE_PREINSTALLED_ABSEIL
+    "Look for a preinstalled version of Protobuf in the system instead of installing a prebuilt binary using FetchContent."
+    OFF
+  )
+
+  # If P4C_USE_PREINSTALLED_ABSEIL is ON just try to find a preinstalled version of Abseil.
+  if(P4C_USE_PREINSTALLED_ABSEIL)
+    if(ENABLE_ABSEIL_STATIC)
+      set(SAVED_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+      set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
+    endif()
+    find_package(absl CONFIG)
+    if(NOT absl_FOUND)
```

Comment:
This was a hack required for bad versions of Protobuf. We might not need it for Abseil.

Reply:
Yeah. Actually we might just use policy instead

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html Looks like this is what is needed :)